src/shimmerHttp.js

```diff
@@ -0,0 +1,215 @@
+      ? data.headers[k].join(' ')
+      : data.headers[k];
+  });
+
+  return excludeUnnecessaryReqKeys(data);
+}
+
+const initialResKeys = ['headers', 'statusCode', 'statusMessage'];
+
+function getResDataObject(res) {
+  return pickBy(res, (v, k) => initialResKeys.indexOf(k) > -1);
+}
+
+const defaultKeysToRecord = [
+  'req.headers.user-agent',
+  'req.headers.accept-encoding',
+  'req.method',
+  'req.path',
+  'req.protocol',
+  'req.port',
+  'req.hostname',
+  'req.hash',
+  'req.pathname',
+  'req.url',
+  'req.query',
+  'req.user-agent',
+  'req.accept-encoding',
+  'res.headers.cache-control',
+  'res.headers.content-type',
+  'res.headers.date',
+  'res.headers.etag',
+  'res.headers.strict-transport-security',
+  'res.headers.content-encoding',
+  'res.headers.content-length',
+  'res.headers.age',
+  'res.headers.connection',
+  'res.headers.server',
+  'res.headers.vary',
+  'res.statusCode',
+  'res.statusMessage'
+];
+
+function filterData(config = {}, obj = {}) {
+  if (typeof config.filter === 'function') {
+    return config.filter(obj);
+  }
+  return pickBy(obj, (v, k) => defaultKeysToRecord.indexOf(k) > -1);
+}
+
+function wrapHttpRequest({ timeline, data: moduleData = {}, config = {} }) {
+  return function wrapper(original) {
+    return function execute(rawOptions, originalCallback) {
+      // bail if we have already started tracking this request
+      // this can happen by calling https.request(opts)
+      // which ends up calling http.request(opts)
+      if (originalCallback && originalCallback.__iopipeTraceId) {
+        return original.apply(this, [rawOptions, originalCallback]);
+      }
+
+      // id of this particular trace
+      const id = uuid();
+      // start the trace
+      timeline.mark(`start:${id}`);
+
+      // setup http trace data that will be sent to IOpipe later
+      moduleData[id] = {};
+      moduleData[id].req = getReqDataObject(rawOptions);
+
+      // the func to execute at the end of the http call
+      function extendedCallback(res) {
+        timeline.mark(`end:${id}`);
+        // add full response data
+        moduleData[id].res = getResDataObject(res);
+        // flatten object for easy transformation/filtering later
+        moduleData[id] = flatten(moduleData[id]);
+        moduleData[id] = filterData(config, moduleData[id]);
+
+        // if filter function returns falsey value, drop all data completely
+        if (typeof moduleData[id] !== 'object') {
+          timeline.data = timeline.data.filter(
+            d => !new RegExp(id).test(d.name)
+          );
+          delete moduleData[id];
+        }
+
+        if (originalCallback) {
+          return originalCallback.apply(this, [res]);
+        }
+        return true;
+      }
+
+      // add traceId to callback so we do not create duplicate data from inner http calls
+      // this can happen for the https module which calls the http module internally
+      extendedCallback.__iopipeTraceId = id;
```

Comment:
If `https` just calls `http` why are we wrapping `https` at all?

Reply:
Let me dig and make sure we need to wrap https